src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/Packaging.targets

```diff
@@ -349,6 +349,10 @@
       <FrameworkReference Condition="'%(InboxOnTargetFramework.AsFrameworkReference)' == 'true'" Include="$(Id)">
         <TargetFramework>$(_target)</TargetFramework>
       </FrameworkReference>
+      <FrameworkReference Condition="'%(InboxOnTargetFramework.AsFrameworkReference)' != 'true' And '%(InboxOnTargetFramework.AsFrameworkReference)' != ''"
```

Comment:
Instead of AsFrameworkReference did you consider just "FrameworkReferences" that way you aren't overloading the metadata?